tests/modules/images/webp-uploads/helper-tests.php

```diff
@@ -270,7 +270,33 @@ function () {
 		$this->assertArrayHasKey( 'filesize', $result );
 		$this->assertArrayHasKey( 'file', $result );
 		$this->assertStringEndsWith( 'image.webp', $result['file'] );
-		$this->assertFileExists( '/tmp/image.webp' );
+	}
+
+	/**
+	 * Tests the webp_uploads_pre_generate_additional_image_source filter returning filesize property.
+	 *
+	 * @test
+	 */
+	public function it_should_use_filesize_when_filter_webp_uploads_pre_generate_additional_image_source_returns_filesize() {
+		remove_all_filters( 'webp_uploads_pre_generate_additional_image_source' );
+
+		add_filter(
+			'webp_uploads_pre_generate_additional_image_source',
+			function () {
+				return array(
+					'file'     => 'image.webp',
+					'filesize' => 777,
+				);
+			}
+		);
+
+		$result = webp_uploads_generate_additional_image_source( 0, 'medium', array(), 'image/webp', '/tmp/image.jpg' );
```

Comment:
Not a blocker, but I'd be in favor of using a real `$attachment_id` and `$size_data` here, similar to the test method right above. Yes, it is strictly not necessary, but I think it's better to use data that is appropriate for what the function expects. Unless we are testing explicitly how it behaves with "weird" data, which is not the case here though.

Reply:
Ok, updated.